src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -574,7 +578,18 @@ perform_multi(CURL *curl, struct curl_slist *headers,
     connection->error = false;
     sol_vector_init(&connection->watches, sizeof(struct connection_watch));
 
-    sol_buffer_init(&connection->buffer);
+    recv_size = interface->recv_buffer_size;
+    if (recv_size != SIZE_MAX) {
+        recv_size = RECV_DEFAULT_SIZE;
+
+        buf = malloc(recv_size);
+        SOL_NULL_CHECK_GOTO(buf, err_buf);
+
+        sol_buffer_init_flags(&connection->buffer, buf, recv_size,
+            SOL_BUFFER_FLAGS_NO_NUL_BYTE | SOL_BUFFER_FLAGS_FIXED_CAPACITY);
+    } else
+        sol_buffer_init_flags(&connection->buffer, NULL, 0,
+            SOL_BUFFER_FLAGS_NO_NUL_BYTE | SOL_BUFFER_FLAGS_DEFAULT);
```

Comment:
this behavior diverges a bit from what we wrote regarding `rx_size`, we better pick one approach and follow it. If we say `4096` then use that instead of allocating using a dynamic buffer.